src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -190,22 +220,95 @@ private async Task<int> StartMonitor()
                 }
                 providerString = sb.ToString();
             }
+            return providerString;
+        }
+        
+
+        private async Task<int> Start(bool liveMonitor)
```

Comment:
Again I'd try to remove needing this argument. Error checks that are unique to export scenario can be hoisted into Export(). processName and _output can be passed to the exporter in the constructor or in properties before entering this call. The processId validation can be refactored into a separate method that is called before this one so that the code to get processName can still benefit from it.